rllib/agents/trainer.py

```diff
@@ -661,6 +662,52 @@ def setup(self, config: PartialTrainerConfigDict):
         # be a partial config dict with the class' default).
         self.config = self.merge_trainer_configs(self._default_config, config,
                                                  self._allow_unknown_configs)
+        framework = self.config["framework"]
+        tf_valid_frameworks = {"tf", "tf2", "tfe"}
+        if framework in tf_valid_frameworks:
+            if not (tf1 or tf):
+                raise ImportError(
+                    ("TensorFlow, was the specified as the 'framework' inside "
+                     "of your config dictionary, however, there was no "
+                     "installation found. You can install tensorflow via "
+                     "pip: pip install tensorflow"))
+        elif framework == "torch":
+            if not torch:
+                raise ImportError(
+                    ("torch, was the specified as the 'framework' inside "
+                     "of your config dictionary, however, there was no "
+                     "installation found. You can install torch via "
+                     "pip: pip install torch"))
+        else:
```

Comment:
This is actually ok. Some users use neither TF nor Torch with RLlib, so they specify e.g. framework=None.

Reply:
fixed